sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -213,6 +215,26 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
   }
 }
 
+/**
+ * Replaces [[OrcDataSourceV2]] with [[DataSource]] if parent node is [[InsertIntoTable]].
+ * This is because [[OrcDataSourceV2]] doesn't support writing data yet.
```

Comment:
Shouldn't Spark handle this already when it sees that OrcDataSourceV2 doesn't implement WriteSupport?

Reply:
Not for `InsertIntoTable`.

Reply:
Is this also a temporary hack, then? It seems like Spark should know it can't write to a source which doesn't implement WriteSupport, no matter what the shape of the query performing the write is.

Reply:
This is a tempory hack. v1 has the same problem: when inserting into a table which is backed by a non-writable data source, Spark would fail during planning.